src/domains/ecommerce-app/api/npg-notification/_npg_notifications_policy.xml.tpl

```diff
@@ -0,0 +1,95 @@
+<policies>
+    <inbound>
+        <base />
+        <!-- start policy variables -->
+        <set-variable name="paymentMethodBackendUri" value="https://${hostname}/pagopa-ecommerce-payment-methods-service" />
+        <set-variable name="transactionServiceBackendUri" value="https://${hostname}/pagopa-ecommerce-transactions-service" />
+        <set-variable name="sessionId" value="@(context.Request.MatchedParameters["sessionId"])" />
+        <set-variable name="paymentMethodId" value="@(context.Request.Url.Query.GetValueOrDefault("paymentMethodId",""))" />
+        <set-variable name="npgNotificationRequestBody" value="@((JObject)context.Request.Body.As<JObject>(true))" />
+        <!-- end policy variables -->
+        <!-- payment method verify session -->
+        <send-request mode="new" response-variable-name="paymentMethodSessionVerificationResponse" timeout="10" ignore-error="true">
+            <set-url>@(String.Format((string)context.Variables["paymentMethodBackendUri"]+"/payment-methods/{0}/sessions/{1}/transactionId", (string)context.Variables["paymentMethodId"], (string)context.Variables["sessionId"]))</set-url>
+            <set-method>GET</set-method>
+            <set-header name="Content-Type" exists-action="override">
+                <value>application/json</value>
+            </set-header>
+            <set-header name="Authorization" exists-action="override">
+                <value> @{
+                        JObject requestBody = (JObject)context.Variables["npgNotificationRequestBody"];
+                        return "Bearer " + (string)requestBody["securityToken"];
+                    }</value>
+            </set-header>
+        </send-request>
+        <choose>
+            <when condition="@(((int)((IResponse)context.Variables["paymentMethodSessionVerificationResponse"]).StatusCode) != 200)">
+                <return-response>
+                    <set-status code="500" reason="Error retrieving session" />
+                </return-response>
+            </when>
+        </choose>
+        <set-variable name="transactionId" value="@(((string)((IResponse)context.Variables["paymentMethodSessionVerificationResponse"]).Body.As<JObject>(preserveContent: true)["transactionId"]))" />
+        <!-- end payment method verify session -->
+        <!-- send transactions service PATCH request -->
+        <send-request mode="new" response-variable-name="transactionServiceAuthorizationPatchResponse" timeout="10" ignore-error="true">
+            <set-url>@(String.Format((string)context.Variables["transactionServiceBackendUri"]+"/transactions/{0}/auth-requests", (string)context.Variables["transactionId"]))</set-url>
```

Comment:
The payment method still doesn't return base64 right?

Reply:
Transaction id will be returned in base64 format by payment methods. It will be implemented in a separate task